pkg/build/builder/dockerutil.go

```diff
@@ -28,7 +29,11 @@ type DockerClient interface {
 	RemoveImage(name string) error
 }
 
-// pushImage pushes a docker image to the registry specified in its tag
+// pushImage pushes a docker image to the registry specified in its tag.
+// The method will retry to push the image when following scenarios occur:
+// - Docker registry is down temporarily or permanently
```

Comment:
@rhcarvalho docker registry service can exists and the Pod can exists as well, but OpenShift might not have capacity to start it... in that case, it is not clear if it is down temporarily or permanently